packages/next/src/build/index.ts

```diff
@@ -1717,12 +1717,16 @@ export default async function build(
             require('next/dist/compiled/glob') as typeof import('next/dist/compiled/glob')
           const glob = (pattern: string): Promise<string[]> => {
             return new Promise((resolve, reject) => {
-              globOrig(pattern, { cwd: dir }, (err, files) => {
-                if (err) {
-                  return reject(err)
+              globOrig(
+                pattern,
+                { cwd: dir, nodir: true, dot: true },
```

Comment:
Might want to add a test for dot files too

Reply:
Added in https://github.com/vercel/next.js/pull/46011/commits/8f7fdb064de18342731b60dc1ebe1c4f9f67db11